unit/testing-utils/free_form_cmdline.cpp

```diff
@@ -0,0 +1,31 @@
+/*******************************************************************\
+
+ Module: Unit test utilities
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include "free_form_cmdline.h"
+
+void free_form_cmdlinet::create_flag(const std::string &flag_name)
+{
+  optiont option;
+  option.isset= true;
+  option.hasval = false;
+  option.islong = false;
+  option.optstring = flag_name;
+  options.push_back(option);
+}
+
+void free_form_cmdlinet::add_flag(std::string flag)
+{
+  create_flag(flag);
+  set(flag);
```

Comment:
Doesn't `create_flag` already set it to true? Maybe `create_flag` can only push it in options but not set it.